utils/error.js

```diff
@@ -0,0 +1,34 @@
+const MISSING_REQUIRED_FIELDS = 1000;
+const MISSING_EMAIL = 1002;
+const MISSING_FULLNAME = 1003;
+const MISSING_USER_ID = 1004;
+const USER_NOT_FOUND = 2001;
+const EMAIL_USERNAME_PASSWORD_INVALID = 3006;
+
+let errorMap = {};
+errorMap[MISSING_REQUIRED_FIELDS] = "Missing required field(s)";
+errorMap[MISSING_EMAIL] = "Missing email";
+errorMap[MISSING_FULLNAME] = "Missing fullname";
+errorMap[MISSING_USER_ID] = "Missing user ID";
+errorMap[USER_NOT_FOUND] = "User not found";
+errorMap[EMAIL_USERNAME_PASSWORD_INVALID] =
+  "Email/Username or password is invalid";
```

Comment:
why don't we construct the object like this  🤣 
```suggestion
const errorMap = {
  [MISSING_REQUIRED_FIELD]: "Missing required field(s)",
  [MISSING_EMAIL]: "Missing email",
  [MISSING_FULLNAME]: "Missing fullname",
  [MISSING_USER_ID]: "Missing user ID",
  [USER_NOT_FOUND]: "User not found",
  [EMAIL_USERNAME_PASSWORD_INVALID]: "Email/Username or password is invalid"
};
```

Reply:
Vì lúc đó cái attribute thành int. Lúc look up ko ra. :D

Reply:
Em sẽ investigate thêm xem có cách nào làm file này nó đẹp không chứ một lần define phải nhập 3 chỗ thì nhìn cũng messy quá.

Reply:
anh nghĩ vẫn ra chứ, chứ cách cũ nó vẫn ra là 1 map giữa `int` và `string` mà hmm?

- Cách cũ
![image](https://user-images.githubusercontent.com/43344276/101233003-6a653280-36f0-11eb-9c8f-daab49a8fd11.png)

- Cách anh suggest
![image](https://user-images.githubusercontent.com/43344276/101233051-bfa14400-36f0-11eb-8266-6750d8cc4f6f.png)

Reply:
nhưng hiện tại có thể để như cũ, không sao, chỉ là anh nhìn thấy kì kì thôi, hơi bias =))

Reply:
Em tính reconstruct lại như thế này 
```
class FcError extends Error {
  constructor(err, data = null) {
    super(err.msg);
    this.code = err.code;
    this.msg = err.msg;
    this.data = data;
  }
}

const ErrorCode = {
  MISSING_REQUIRED_FIELDS: {code: 1000, msg: "Missing required field(s)"}
}
``` 


Có nghĩ mỗi err code mới mình define 1 chỗ thôi. Em nghĩ sẽ tốt hơn nếu em để ở 1 cái PR khác. This PR is quite large.

Reply:
cc: @ndkhaivn Kh